docs/concepts/glossary.md

```diff
@@ -0,0 +1,166 @@
+---
+title: Glossary
+nav-pos: 3
+nav-title: Glossary
+nav-parent_id: concepts
+---
+<!--
+Licensed to the Apache Software Foundation (ASF) under one
+or more contributor license agreements.  See the NOTICE file
+distributed with this work for additional information
+regarding copyright ownership.  The ASF licenses this file
+to you under the Apache License, Version 2.0 (the
+"License"); you may not use this file except in compliance
+with the License.  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+
+#### Flink Application Cluster
```

Comment:
Personally, I call these `job cluster` in line with the internals, e.g. [JobClusterEntrypoint](https://github.com/apache/flink/blob/master/flink-runtime/src/main/java/org/apache/flink/runtime/entrypoint/JobClusterEntrypoint.java).

cc @tillrohrmann

Reply:
It is the same for me. I'm also fine with changing the terminology as long as we don't confuse our users.

Reply:
The term "Flink Application (Cluster)" was proposed by @StephanEwen to emphasize, that you basically run a Flink job like an classical application in this mode (or at least this is the mid-term goal (passive resource manager, Flink as a library). 

@tillrohrmann @uce Do you think we can change this at this point?

Reply:
I do like the proposed name and I'd be happy to call it application cluster going forward myself.

I think having slight misalignment with internal class naming is acceptable, but I'd suggest to update user facing references to job cluster if we keep the renaming:
* https://ci.apache.org/projects/flink/flink-docs-release-1.8/ops/deployment/kubernetes.html#flink-job-cluster-on-kubernetes
* https://github.com/apache/flink/tree/master/flink-container/docker
* https://github.com/apache/flink/tree/master/flink-container/kubernetes

Otherwise I would be afraid of ending up in a situation with inconsistent names which can be confusing to users.

Since I am not involved in the day to day development of this component, I would leave the decision to @tillrohrmann whether we can change it at this point.

Reply:
Agreed. There are actually a few changes "planned" to this module anyway (see discussion of https://issues.apache.org/jira/browse/FLINK-12546), so this might actually be good timing to rename it.

Reply:
@tillrohrmann is fine to change this. @uce I created https://issues.apache.org/jira/browse/FLINK-12885 to rename Job Cluster to Application Cluster. Shouldn't be a problem to do this before the 1.9 release.

Reply:
I have already a local version of FLINK-12885 now, but it is still blocked on a different issue (FLINK-12752). Once this is merged, we can align this pretty quickly.